Source/WebCore/animation/StyleOriginatedAnimation.cpp

```diff
@@ -237,21 +237,25 @@ AnimationEffectPhase StyleOriginatedAnimation::phaseWithoutEffect() const
     return *animationCurrentTime < 0_s ? AnimationEffectPhase::Before : AnimationEffectPhase::After;
 }
 
-Seconds StyleOriginatedAnimation::effectTimeAtStart() const
+CSSNumberishTime StyleOriginatedAnimation::effectTimeAtStart() const
 {
     if (auto* effect = this->effect())
         return effect->delay();
     return 0_s;
 }
 
-Seconds StyleOriginatedAnimation::effectTimeAtIteration(double iteration) const
+CSSNumberishTime StyleOriginatedAnimation::effectTimeAtIteration(double iteration) const
 {
-    if (auto* effect = this->effect())
-        return effect->delay() + effect->iterationDuration() * iteration;
+    if (auto* effect = this->effect()) {
+        auto iterationDuration = effect->iterationDuration();
+        if (iterationDuration.percentage())
+            return iterationDuration;
+        return effect->delay() + iterationDuration * iteration;
```

Comment:
Can you add a comment explaining why it's OK not to add the delay or multiply by the iteration number ? (The comment could be as simple as a link to the spec on top of the function definition)

Not super clear from the code atm.

Reply:
Will do! At the moment there is no support for delay for progress-based animations as specified. Will make a note of that in the relevant places.